### PR TITLE
stake distribution calculation

### PR DESCRIPTION
**Pull request by JaredCorduan**
This adds the stake distribution calculation to the latex spec. #118

### CODE REVIEW

File: latex/epoch.tex [hunk 1]
```diff
@@ -77,6 +89,65 @@
...
 
+%%
+%% Figure - Functions for Stake Distribution
+%%
+\begin{figure}[htb]
+  \begin{align*}
+      & \fun{baseStake} \in \powerset{\TxOut} \to \powerset{(\HashKey \times \Coin)}
+      & \text{base address stake} \\
+      & \fun{baseStake}~{txouts} = \\
+      & ~~ \{(hk, c) \mid a\in\Addr_{base} \land (a, c) \in txouts \land \fun{stakeHK}~a = hk \}
```
Comment (polinavino) [inline latex/epoch.tex]: is it ok to call P(TxOut) txouts? It might be easy to confuse with the txouts tx, which gives a UTxO?

Comment (JaredCorduan) [reply latex/epoch.tex]: yea, that could be confusing. the variables are in italics, and the functions are in mathsf, so you can tell the difference, but it is still probably confusing. maybe `outs`?

File: latex/epoch.tex [hunk 2]
```diff
@@ -77,6 +89,65 @@
...
+      \nextdef
+      & \fun{ptrStake} \in \powerset{\TxOut} \to (\Ptr \mapsto \HashKey)
+          \to \powerset{(\HashKey \times \Coin)}
+      & \text{base address stake} \\
+      & \fun{ptrStake}~{txouts}~{ptrs} = \\
```
Comment (polinavino) [inline latex/epoch.tex]: this one is should be ptr stake i think? in the \text

File: latex/epoch.tex [hunk 3]
```diff
@@ -77,6 +89,65 @@
...
+      & ~~ \left\{
+             hk\mapsto \sum_{hk\mapsto c}c
+             ~\Big\vert~
+             \begin{array}{r}
+             (\isActive{hk}{stkeys}{delegs}{stpools}) \\
+             \land (hk, c) \in (\fun{stake}~{txouts}~{ptrs})
+             \end{array}
+           \right\}
+      \nextdef
+      & \fun{totalStake} \in \powerset{\TxOut} \to (\Ptr \mapsto \HashKey) \to \Allocs\\
```
Comment (polinavino) [inline latex/epoch.tex]: in "(hk, c) ∈ (stake txouts ptrs)", hk is a hash key, and the result type is VKey \to Coin
How should we reconcile this?

File: latex/epoch.tex [hunk 4]
```diff
@@ -37,11 +39,21 @@
...
 \begin{figure}[htb]
-  \emph{Abstract types}
+  \emph{Derived types}
   %
   \begin{equation*}
-    \begin{array}{r@{~\in~}ll}
-      \var{production} & \Production & \text{blocks produced in the epoch}\\
+    \begin{array}{r@{~\in~}lr}
+      hk \mapsto n & \Production & \text{blocks produced in the epoch}
+    \end{array}
```
Comment (polinavino) [inline latex/epoch.tex]: is Production : VKey \to N? Or HashKey \to N?
Also Production is just the number of blocks associated to this key in this epoch?
Is it not used at all anywhere, not even the stakeDistr calculation?

File: latex/epoch.tex (hunk 1 above)
Comment (polinavino) [reply latex/epoch.tex]: i think that would work

File: latex/epoch.tex (hunk 4 above)
Comment (JaredCorduan) [reply latex/epoch.tex]: yes, it's not used yet, but it is about to be used for the rewards calculation. I probably should have not included this in this PR.

File: latex/epoch.tex [hunk 5]
```diff
@@ -77,6 +89,65 @@
...
+      & \fun{stake} \in \powerset{\TxOut} \to (\Ptr \mapsto \HashKey)
+          \to \powerset{(\HashKey \times \Coin)}
+      & \text{pointer address stake} \\
+      & \fun{stake}~{outs}~{ptrs} = (\fun{baseStake}~{outs}) \cup (\fun{ptrStake}~{outs}~{ptrs})
+      \nextdef
+      & \fun{isActive} \in \HashKey \to \Allocs \to (\HashKey \mapsto \HashKey) \to \Allocs \to \mathbb{B}
+      & \text{active stake} \\
+      & \isActive{v_{sk}}{stkeys}{delegs}{stpools} = \exists \var{v_{p}}\\
+      & ~~ \var{v_{sk}} \in \dom stkeys
+              \land \var{v_{st}}\mapsto\var{v_{p}} \in delegs
```
Comment (polinavino) [inline latex/epoch.tex]: v_{sk}?

File: latex/epoch.tex [hunk 6]
```diff
@@ -77,6 +89,65 @@
...
+      & \fun{baseStake} \in \powerset{\TxOut} \to \powerset{(\HashKey \times \Coin)}
+      & \text{base address stake} \\
+      & \fun{baseStake}~{outs} = \\
+      & ~~ \{(hk, c) \mid a\in\Addr_{base} \land (a, c) \in outs \land \fun{stakeHK}~a = hk \}
+      \nextdef
+      & \fun{ptrStake} \in \powerset{\TxOut} \to (\Ptr \mapsto \HashKey)
+          \to \powerset{(\HashKey \times \Coin)}
+      & \text{base address stake} \\
+      & \fun{ptrStake}~{outs}~{ptrs} = \\
+      & ~~ \{(hk, c) \mid a\in\Addr_{ptr} \land (a, c) \in outs \land (\fun{stakePtr}~a\mapsto hk \in ptrs \}
```
Comment (polinavino) [inline latex/epoch.tex]: missing bracket i think

File: latex/epoch.tex [hunk 7]
```diff
@@ -77,6 +89,65 @@
...
+      \nextdef
+      & \fun{stake} \in \powerset{\TxOut} \to (\Ptr \mapsto \HashKey)
+          \to \powerset{(\HashKey \times \Coin)}
+      & \text{pointer address stake} \\
```
[the remark this answers is not in the document]
Comment (JaredCorduan) [reply latex/epoch.tex]: thanks! I'll call it "potential stake".

**Pull request merged**